library/spdm_responder_lib/libspdm_rsp_algorithms.c

```diff
@@ -383,7 +383,16 @@ libspdm_return_t libspdm_get_response_algorithms(void *context,
                      sizeof(spdm_negotiate_algorithms_request_t) +
                      sizeof(uint32_t) * spdm_request->ext_asym_count +
                      sizeof(uint32_t) * spdm_request->ext_hash_count);
+        uint8_t alg_type_pre = struct_table->alg_type;
```

Comment:
This declaration needs to be at line 381. Even though libspdm is a C99 project we try to make it as compatible with C89 as possible.